tensorflow/contrib/rnn/python/ops/rnn_cell.py

```diff
@@ -1926,3 +1927,158 @@ def call(self, inputs, state):
+        in order to reduce the scale of forgetting at the beginning of
+        the training.
+      activation: Activation function of the inner states.
+    """
+    self._num_units = num_units
+    self._initializer = initializer
+    self._num_proj = num_proj
+    self._forget_bias = forget_bias
+    self._activation = activation
+    self._number_of_groups = number_of_groups
+
+    assert (self._num_units % self._number_of_groups == 0)
+    if self._num_proj:
+      assert (self._num_proj % self._number_of_groups == 0)
+      self._group_shape = [int(self._num_proj / self._number_of_groups),
+                           int(self._num_units / self._number_of_groups)]
+    else:
+      self._group_shape = [int(self._num_units / self._number_of_groups),
+                           int(self._num_units / self._number_of_groups)]
+
+    if num_proj:
+      self._state_size = (core_rnn_cell.LSTMStateTuple(num_units, num_proj))
+      self._output_size = num_proj
+    else:
+      self._state_size = (core_rnn_cell.LSTMStateTuple(num_units, num_units))
+      self._output_size = num_units
+
+  @property
+  def state_size(self):
+    return self._state_size
+
+  @property
+  def output_size(self):
+    return self._output_size
+
+  def _get_input_for_group(self, inpt, group_id, group_size):
+    """Slices inputs into groups to prepare for processing by cell's groups
+
+    Args:
+      inpt: inputs
+      group_id: group id, for which to prepare extract input_group_id
+      group_size: size of the group
+
+    Returns:
+      subset of inputs corresponding to group "group_id"
+    """
+    return array_ops.slice(input_=inpt,
+                           begin=[0, group_id * group_size],
+                           size=[inpt.get_shape()[0].value, group_size],
+                           name="GLSTMinputGroupCreation")
+
+  def __call__(self, inputs, state):
+    """Run one step of G-LSTM.
+
+    Args:
+      inputs: input Tensor, 2D, batch x num_units.
+      state: this must be a tuple of state Tensors, both `2-D`, with column sizes `c_state` and `m_state`.
+
+    Returns:
+      A tuple containing:
+
+      - A `2-D, [batch x output_dim]`, Tensor representing the output of the
+        G-LSTM after reading `inputs` when previous state was `state`.
+        Here output_dim is:
+           num_proj if num_proj was set,
+           num_units otherwise.
+      - Tensor(s) representing the new state of G-LSTM after reading `inputs` when
+        the previous state was `state`.  Same type and shape(s) as `state`.
+
+    Raises:
+      ValueError: If input size cannot be inferred from inputs via
+        static shape inference.
+    """
+    (c_prev, m_prev) = state
+
+    input_size = inputs.get_shape().with_rank(2)[1]
+    if input_size.value is None:
+      raise ValueError("Could not infer input size from inputs.get_shape()[-1]")
+    dtype = inputs.dtype
+    with vs.variable_scope("glstm_cell",
```

Comment:
with vs.variable_scope(vs.get_variable_scope(), initializer=...):

(the parent `__call__` already sets the current variable scope to "glstm_cell")